/*
 * Copyright (C) 2019 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package apex.proto;

message SessionState {

  enum State {
    UNKNOWN = 0;
    VERIFIED = 1;
    STAGED = 2;
    ACTIVATED = 3;
    ACTIVATION_FAILED = 4;
    SUCCESS = 5;
    ROLLBACK_IN_PROGRESS = 6;
    ROLLED_BACK = 7;
    ROLLBACK_FAILED = 8;
  }

  int32 id = 1;

  // Current state of the session
  State state = 2;

  // Child session ids
  repeated int32 child_session_ids = 3;
}
